flink-core/src/main/java/org/apache/flink/api/common/state/InternalCheckpointListener.java

```diff
@@ -0,0 +1,52 @@
+/*
+
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+
+*/
+
+package org.apache.flink.api.common.state;
+
+import org.apache.flink.annotation.Internal;
+
+/**
+ * The internal checkpoint listener add another {@link #notifyCheckpointSubsumed(long)} interface
+ * for internal usage.
+ */
+@Internal
+public interface InternalCheckpointListener extends CheckpointListener {
```

Comment:
Do we need this interface in the first place? The only value I can see of this is to have a common signature of this `notifyCheckpointSubsumed()`, which maybe is a good reason on it's own, which would make refactoring/renaming easier, but I'm not entirely sure...

Also I'm not entirely sure if `InternalCheckpointListener` should extend `CheckpointListener` or rather we should have another smaller interface `CheckpointSubsumedListener` that just adds this single method.

Reply:
My original idea is to add the new method in `CheckpointListener`. However, current `CheckpointListener` is already annotated as `public` and the semantics of a checkpoint subsumed should be acted as an internal operation only for runtime classes.
Maybe we can see the relationship as `InternalKvState` and `State`.